NetworkLayer/Network/AdCategoriesService.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  AdCategoriesService.swift
+//  NetworkLayer
+//
+//  Created by Pierre on 15/11/2022.
+//
+
+import Foundation
+
+class AdCategoriesService {
+    let session: URLSession
+    
+    var urlString = "https://raw.githubusercontent.com/leboncoin/paperclip/master/categories.json"
+    
+    init(urlSession: URLSession = .shared) {
```

Comment:
```suggestion
    init(urlSession: URLSession = URLSession(configuration: .ephemeral)) {
```

Le .shared est une configuration par défaut qui de base conserve des éléments en cache, ici ça ne nous est d'aucune utilité. On préférera utiliser une configuration ephemeral pour ne rien garder en mémoire.